Tests/Unit/LoraKeysManagerFacade/SendCloudToDeviceMessageTest.cs

```diff
@@ -366,6 +385,47 @@ public async Task When_Querying_Devices_And_Finds_Single_Gateway_Class_C_Should_
             query.VerifyAll();
         }
 
+        [Fact]
+        public async Task When_Querying_Devices_And_Finds_No_Gateway_For_Class_C_Should_Return_InternalServerError()
+        {
+            const string devEUI = "0123456789";
+
+            var deviceTwin = new Twin
+            {
+                Properties = new TwinProperties()
+                {
+                    Desired = new TwinCollection($"{{\"DevAddr\": \"03010101\", \"ClassType\": \"C\"}}"),
+                    Reported = new TwinCollection(),
+                }
+            };
+
+            var query = new Mock<IQuery>(MockBehavior.Strict);
+            query.Setup(x => x.HasMoreResults).Returns(true);
+            query.Setup(x => x.GetNextAsTwinAsync())
+                .ReturnsAsync(new[] { deviceTwin });
+
+            this.registryManager.Setup(x => x.CreateQuery(It.IsNotNull<string>(), It.IsAny<int?>()))
+                .Returns(query.Object);
+
+            var actualMessage = new LoRaCloudToDeviceMessage()
+            {
+                Fport = TestPort,
+                Payload = "hello",
+            };
+
+            var actual = await this.sendCloudToDeviceMessage.SendCloudToDeviceMessageImplementationAsync(
+                devEUI,
+                actualMessage);
+
+            Assert.IsType<ObjectResult>(actual);
```

Comment:
[nit] we can use the return value of the assertion to avoid a cast on the next line